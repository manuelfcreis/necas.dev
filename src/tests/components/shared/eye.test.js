import React from "react";
import Eye from "../../../components/shared/eye";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const element = shallow(
    <Eye small={false} positions={{ x: 50, y: 50 }} hide={false} />
  );
  expect(element).toMatchSnapshot();
});

it("calculates eye position correctly", () => {
  const wrapper = shallow(
    <Eye small={false} positions={{ x: 50, y: 50 }} hide={false} />
  );

  expect(wrapper.instance().calculateEyePosition(1, 1, "top")).toEqual("80%");
});
