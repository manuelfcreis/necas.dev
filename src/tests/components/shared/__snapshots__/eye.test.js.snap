// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="eye"
>
  <div
    className="ball"
    style={
      Object {
        "left": "80%",
        "top": "80%",
      }
    }
  >
    <div
      className="iris"
    />
  </div>
</div>
`;
